swagger.yaml

```diff
@@ -1,75 +1,164 @@
+  - "application/json"
+
 paths:
-  /:
+
+  /permissions/{osuID}:
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      tags:
+      - "permissions"
+      summary: "Retrieve permissions for user by their OSU ID"
+      operationId: "getPermissionsById"
       parameters:
-        - $ref: '#/parameters/pretty'
+        - name: "osuID"
+          in: "path"
+          description: "OSU ID of person whose permissions will be retrieved"
+          required: true
+          type: "string"
+
       responses:
         200:
-          description: Object containing information
+          description: "Permissions successfully retrieved"
           schema:
-            $ref: '#/definitions/Info'
+            $ref: "#/definitions/PermissionsResult"
+        404:
+          description: "No permissions found for this OSU ID"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
-parameters:
-  pretty:
-    name: pretty
-    in: query
-    type: boolean
-    required: false
-    description: If true, JSON response will be pretty-printed
+  /permissions:
+    get:
+      tags:
+      - "permissions"
+      summary: "Retrieve permissions for users by parameters"
+      operationId: "getPermissions"
+      parameters:
+        - name: "osuID"
+          in: "query"
+          description: "OSU ID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+        - name: "onid"
+          in: "query"
+          description: "ONID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+      responses:
+        200:
+          description: "Permissions successfully retrieved"
+          schema:
+            $ref: "#/definitions/PermissionsResultArray"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
 
 definitions:
-  Info:
+
+  PermissionLevel:
+    type: "integer"
+    minimum: 0
+    maximum: 5
+    description: "Permission level for a given category"
+    example: 3
+
+  Permissions:
+    type: "object"
+    properties:
+      onid:
+        type: "string"
+        description: "ONID of person who permissions belong to"
+        example: "browni"
+      hrPermissions:
+        $ref: "#/definitions/PermissionLevel"
+      studentPermissions:
+        $ref: "#/definitions/PermissionLevel"
+      financePermissions:
+        $ref: "#/definitions/PermissionLevel"
+
+  SelfLink:
+    type: "object"
+    properties:
+      self:
+        type: "string"
+        format: "url"
+        description: "Self link of resource"
+        example: "https://api.oregonstate.edu/api/v1/permissions/123456789"
```

Comment:
For APIs going through the gateway (Apigee), the url doesn't have `api` in it. So just `"https://api.oregonstate.edu/v1/permissions/123456789"`